src/NzbDrone.Core.Test/MediaFiles/EpisodeImport/ImportDecisionMakerFixture.cs

```diff
@@ -226,10 +233,100 @@ public void should_not_throw_if_episodes_are_not_found()
+        }
+
+        [Test]
+        public void should_not_use_folder_for_full_season()
+        {
+            var videoFiles = new[]
+                             {
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E01.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E02.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01\S01E03.mkv".AsOsAgnostic()
+                             };
+
+            GivenSpecifications(_pass1);
+            GivenVideoFiles(videoFiles.ToList());
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Exactly(3));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.Is<ParsedEpisodeInfo>(p => p != null)), Times.Never());
+        }
+
+        [Test]
+        public void should_not_use_folder_when_it_contains_more_than_one_valid_video_file()
+        {
+            var videoFiles = new[]
+                             {
+                                 @"C:\Test\Unsorted\Series.Title.S01E01\S01E01.mkv".AsOsAgnostic(),
+                                 @"C:\Test\Unsorted\Series.Title.S01E01\1x01.mkv".AsOsAgnostic()
+                             };
+
+            GivenSpecifications(_pass1);
+            GivenVideoFiles(videoFiles.ToList());
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01E01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Exactly(2));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.Is<ParsedEpisodeInfo>(p => p != null)), Times.Never());
+        }
+
+        [Test]
+        public void should_use_folder_when_only_one_video_file()
+        {
+            GivenSpecifications(_pass1);
+
+            var folderInfo = Parser.Parser.ParseTitle("Series.Title.S01E01");
+
+            Subject.GetImportDecisions(_videoFiles, _series, true, folderInfo);
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, It.IsAny<ParsedEpisodeInfo>()), Times.Exactly(1));
+
+            Mocker.GetMock<IParsingService>()
+                  .Verify(c => c.GetLocalEpisode(It.IsAny<string>(), It.IsAny<Series>(), true, null), Times.Never());
+        }
+
+        [Test]
+        public void should_use_folder_when_only_one_video_file_and_a_sample()
```

Comment:
I think this particular case was an exception, not the rule, every other occurrence has been a dir fix that corrected a misnumbered episode, extra character in the name (S01E143 instead of S01E14) or in the case of Castle a dir fix could have nuked the S07E13 release and fixed it to S07E13.

If I grabbed the S07E03 release and ended up nuking my S07E13 file I'd be in the same boat.

I don't think rejecting them and letting the user choose is right either, in most cases they'd just want to hit import anyways, but which would we choose to follow automatically when that button is pressed, the folder name or the file name? Feels like a 99% feature, 99% of releases would benefit from this (or otherwise be unaffected), the 1% that has issues would need some manual intervention.